components/camel-hazelcast/src/main/java/org/apache/camel/component/hazelcast/topic/HazelcastTopicProducer.java

```diff
@@ -41,19 +42,18 @@ public HazelcastTopicProducer(HazelcastInstance hazelcastInstance, HazelcastDefa
     }
 
     public void process(Exchange exchange) throws Exception {
-        final int operation = lookupOperationNumber(exchange);
+        final HazelcastOperation operation = lookupOperation(exchange);
 
-        switch (operation) {
-        case -1:
-            // default operation to publish
-        case HazelcastConstants.PUBLISH_OPERATION:
+        switch ((operation != null) ? operation : HazelcastOperation.PUBLISH) {
```

Comment:
Yes so that it is configurable and if not operation is provided and default is not set we should throw an exception so that the behavior is always in user control.